android/app/src/main/java/app/organicmaps/bookmarks/BookmarksListFragment.java

```diff
@@ -227,7 +227,6 @@ public void onStop()
   private void configureBookmarksListAdapter()
   {
     BookmarkListAdapter adapter = getBookmarkListAdapter();
-    adapter.registerAdapterDataObserver(mCategoryDataSource);
```

Comment:
This observer updates the cached (basically bookmarkCategory object) with core (by calling onChanged() in dataObserver class) whenever the we notify the adapter that dataset has been changed using the method adapter.notifyDatasetChanged().

Since you removed this observer it may crash the app whenever updated data set was required other than onDelete().

I think that adapter should not be removed since it is called whenever we use the notifyDatasetChanged() method and its the easiest way to update the cached data.